src/sagas/reown.js

```diff
@@ -0,0 +1,996 @@
+            data: request.data,
+            dapp: requestMetadata,
+          }
+        });
+      } break;
+
+      case TriggerTypes.SendNanoContractTxLoadingTrigger:
+        dispatch(setNewNanoContractStatusLoading());
+        dispatch(showGlobalModal(MODAL_TYPES.NANO_CONTRACT_FEEDBACK, { isLoading: true }));
+        resolve();
+        break;
+
+      case TriggerTypes.CreateTokenLoadingTrigger:
+        log.debug('CreateTokenLoadingTrigger: Starting token creation process');
+        dispatch(setCreateTokenStatusLoading());
+        dispatch(showGlobalModal(MODAL_TYPES.TOKEN_CREATION_FEEDBACK, { isLoading: true }));
+        resolve();
+        break;
+
+      case TriggerTypes.CreateTokenLoadingFinishedTrigger:
+        log.debug('CreateTokenLoadingFinishedTrigger: Token creation process completed');
+        dispatch(setCreateTokenStatusReady());
+        dispatch(hideGlobalModal());
+        resolve();
+        break;
+
+      case TriggerTypes.SendNanoContractTxLoadingFinishedTrigger:
+        dispatch(setNewNanoContractStatusReady());
+        resolve();
+        break;
+
+      case TriggerTypes.PinConfirmationPrompt: {
+        const pinPromise = new Promise((pinResolve, pinReject) => {
+          dispatch(showGlobalModal(MODAL_TYPES.PIN_PAD, {
+            onComplete: (pinCode) => {
+              dispatch(hideGlobalModal());
+              pinResolve(pinCode);
+            },
+            onCancel: () => {
+              dispatch(hideGlobalModal());
+              dispatch(setCreateTokenStatusReady()); // Reset loading state if PIN entry is cancelled
+              pinReject(new Error('PIN entry cancelled'));
+            }
+          }));
+        });
+
+        try {
+          const pin = await pinPromise;
+          resolve({
+            type: TriggerResponseTypes.PinConfirmationResponse,
+            data: {
+              accepted: true,
+              pinCode: pin,
+            }
+          });
+        } catch (error) {
+          log.error('PIN confirmation error:', error);
+          reject(error);
+        }
+      } break;
+
+      default:
+        reject(new Error('Invalid request'));
+    }
+  });
+
+/**
+ * Generic handler for dApp requests that require user confirmation
+ * Shows a modal to the user and manages the accept/reject flow
+ * 
+ * @param {Object} payload - The request payload
+ * @param {Function} payload.accept - Callback for when user accepts
+ * @param {Function} payload.deny - Callback for when user rejects
+ * @param {Object} payload.data - The request data
+ * @param {Object} payload.dapp - Metadata about the requesting dApp
+ * @param {string} modalType - The type of modal to show from ReownModalTypes
+ */
+export function* handleDAppRequest({ payload }, modalType) {
+  const { accept, deny: denyCb, data, dapp } = payload;
+  const wallet = getGlobalWallet();
+
+  if (!wallet.isReady()) {
+    log.error('Got a session request but wallet is not ready.');
+    return;
+  }
```

Comment:
What happens with this request? Is it lost? Should we wait for the wallet readiness?